src/main/java/com/gard/testbed/activitySystem/State.java

```diff
@@ -16,21 +16,32 @@ public State(String stateId) {
     }
 
     @Override
-    public String getId() {
-        return id;
+    public boolean isCompleted(IBlackboard player) {
+        return transitions.stream().allMatch(e -> e.isComplete(player));
+    }
+
+    @Override
+    public boolean isActive() {
+        return false;
     }
```

Comment:
The FSM defines your space, your position is an ID on the space. Store the state Id as activity progress. It is not in the activity but in the execution